### PR TITLE
rgw/multisite_reshard: pass the right generation number for bilog listing

### PR DESCRIPTION
**Pull request by smanjara**
passes `test_bucket_reshard_incremental`

Signed-off-by: Shilpa Jagannath <smanjara@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/rgw_data_sync.cc [hunk 1]
```diff
@@ -5318,7 +5322,7 @@ int RGWSyncBucketCR::operate(const DoutPrefixProvider *dpp)
 
         yield call(new RGWSyncBucketShardCR(sc, data_lease_cr, sync_pair,
                                             sync_pipe, bucket_status.state,
-                                            tn, progress));
+                                            *gen, tn, progress));
```
Comment (smanjara) [inline src/rgw/rgw_data_sync.cc:5325]: @cbodley ok to deref gen here?

Comment (cbodley) [reply src/rgw/rgw_data_sync.cc:5323]: no, it looks like we should use `bucket_status.incremental_gen` here instead?

if we get a `gen`, the `if (gen) {` block above will compare it with `bucket_status.incremental_gen` and return if they don't match

Comment (smanjara) [reply src/rgw/rgw_data_sync.cc:5323]: @cbodley ok but passing `bucket_status.incremental_gen` doesn't help fix the issue though.

Comment (smanjara) [reply src/rgw/rgw_data_sync.cc:5323]: hmm it works in both cases actually.

**Review by cbodley — APPROVED** [review]

**Pull request merged**